### PR TITLE
MOIA-31842: Allow empty Strings

### PR DESCRIPTION
**Pull request by saeltz**
_DynamoDB_ does support empty `String`s now:
https://aws.amazon.com/about-aws/whats-new/2020/05/amazon-dynamodb-now-supports-empty-values-for-non-key-string-and-binary-attributes-in-dynamodb-tables/?nc1=h_ls
(Except for keys.)

This allows empty `String`s and updates the tests and documentation.

### CODE REVIEW

File: src/main/scala/scynamo/ScynamoEncoder.scala [hunk 1]
```diff
@@ -26,18 +26,14 @@ object ScynamoEncoder extends DefaultScynamoEncoderInstances {
...
 trait DefaultScynamoEncoderInstances extends ScynamoIterableEncoder {
-  implicit val stringEncoder: ScynamoEncoder[String] = value =>
-    if (value.nonEmpty) Right(AttributeValue.builder().s(value).build())
-    else Either.leftNec(ScynamoEncodeError.invalidEmptyValue(ScynamoType.String))
+  implicit val stringEncoder: ScynamoEncoder[String] = value => Right(AttributeValue.builder().s(value).build())
 
   implicit val stringOptionEncoder: ScynamoEncoder[Option[String]] = {
     case Some("") | None => Right(AttributeValue.builder().nul(true).build())
     case Some(value)     => stringEncoder.encode(value)
   }
```
Comment (joroKr21) [inline src/main/scala/scynamo/ScynamoEncoder.scala:34]: In that case we can remove this
```suggestion
```

**Review by joroKr21 — CHANGES_REQUESTED** [review]
It would be good to also add a test with an empty string attribute.

Empty strings everywhere!

everywhere

File: src/main/scala/scynamo/ScynamoEncoder.scala (hunk 1 above)
Comment (saeltz) [reply src/main/scala/scynamo/ScynamoEncoder.scala:34]: I think this is still valuable.

Comment (joroKr21) [reply src/main/scala/scynamo/ScynamoEncoder.scala:34]: But why, the sole purpose of it was to avoid the error on empty string?

Comment (saeltz) [reply src/main/scala/scynamo/ScynamoEncoder.scala]: Alright. Removed in https://github.com/moia-dev/scynamo/pull/136/commits/d1fdf899f5d96c9bab7bd6634822e4025e3b8bb8.

**Review by joroKr21 — APPROVED** [review]
